default/python/universe_generation/monsters.py

```diff
@@ -1,8 +1,82 @@
 import random
 import freeorion as fo
 import util
+from util import MapGenerationError
 import statistics
 import universe_tables
+from galaxy import DisjointSets
+
+
+class StarlaneAlteringMonsters(object):
+    def __init__(self, systems):
+        self.systems = systems
+        self.placed = set()
+        self.removed_lanes = set()
+        self.starlanes = [(s1, s2) for s1 in systems for s2 in fo.sys_get_starlanes(s1) if s1 < s2]
+
+    def can_place_at(self, system, plan):
+        # Check if the starlane altering monster fleet ''plan'' can be placed at ''system''
+        # without disjoining the galaxy map.
+        # Compute the disjoint set of the galaxy without the starlanes
+        # from the proposed system.  Return False if there will be more
+        # connected regions than the number of placed starlane altering
+        # monsters plus one, otherwise True.
```

Comment:
That should probably turned into docstring format. My bad, have been very sloppy with comments when I put together my alternative implementation.